kura/org.eclipse.kura.core/src/main/java/org/eclipse/kura/core/linux/util/LinuxProcessUtil.java

```diff
@@ -351,4 +393,22 @@ public static String getInputStreamAsString(InputStream stream) throws IOExcepti
         }
         return sb.toString();
     }
+    
+    public static boolean waitProcess(int pid, long poll, long timeout) {
+        boolean exists = false;
+        try {
+            final long startTime = System.currentTimeMillis();
+            long now;
+            do {
+                Thread.sleep(poll);
+                exists = (getPid(pid) >= 0) ? true : false;
```

Comment:
maybe we could use the code in PppLinux and have `exists = LinuxProcessUtil.stop(pid);`